aiutils/data/data.py

```diff
@@ -0,0 +1,87 @@
+"""Utility functions for making batch generation easier."""
+import numpy as np
+from multiprocessing import Process, Queue
+
+
+def sequential(batch_size, num_samples, num_epochs=1):
+    """Generate sequence indices.
+    """
+    for epoch in range(num_epochs):
+        indices = np.arange(num_samples)
+        for i in range(0, num_samples - batch_size + 1, batch_size):
+            slice_ = slice(i, i + batch_size)
+            yield indices[slice_]
+
+
+def random(batch_size, num_samples, num_epochs=1):
+    """Generate random indices.
+    """
+    for epoch in range(num_epochs):
+        indices = np.random.permutation(num_samples)
+        for i in range(0, num_samples - batch_size + 1, batch_size):
+            slice_ = slice(i, i + batch_size)
+            yield indices[slice_]
+
+
+def batch_generator(data, index_generator, batch_function=None):
+    """Generate batches of data.
+    """
+    for samples in index_generator:
+        batch = data.get(samples)
```

Comment:
Probably you were going to do it already but make sure user knows that this function expects data to have get() function